sqlglot/parser.py

```diff
@@ -1595,30 +1591,38 @@ def _parse_create_sequence(self, replace, properties) -> exp.Create | exp.Comman
                 seq.set("data_type", self._parse_types())
             elif self._match_text_seq("INCREMENT"):
                 self._match_text_seq("BY")
+                self._match_text_seq("=")
                 seq.set("increment", self._parse_term())
             elif self._match_text_seq("MINVALUE"):
                 seq.set("minvalue", self._parse_term())
             elif self._match_text_seq("MAXVALUE"):
                 seq.set("maxvalue", self._parse_term())
             elif self._match(TokenType.START_WITH) or self._match_text_seq("START"):
+                self._match_text_seq("=")
                 seq.set("start", self._parse_term())
             elif self._match_text_seq("CACHE"):
                 # T-SQL allows empty CACHE which is initialized dynamically
-                seq.set("cache", self._parse_term() or exp.Var(this=""))
+                cache: t.Optional[exp.Expression | bool] = self._parse_term()
+                if cache and not isinstance(cache, exp.Literal):
+                    self._retreat(self._index - 1)
+                    cache = True
+                seq.set("cache", cache)
             elif self._match_text_seq("OWNED", "BY"):
-                owned = None if self._match_text_seq("NONE") else self._parse_column()
+                # "OWNED BY NONE" is the default
+                owned = self._prev.text if self._match_text_seq("NONE") else self._parse_column()
                 seq.set("owned", owned)
-            elif self._match_text_seq("COMMENT", "="):
-                seq.set("comment", self._parse_string())
+            elif self._match_text_seq("COMMENT"):
+                seq.set("comment", self._parse_property_assignment(exp.SchemaCommentProperty))
             elif self._match_text_seq("SHARING", "="):
                 seq.set(
-                    "sharing",
-                    self._prev.text if self._match_texts(("METADATA", "DATA", "NONE")) else None,
+                    "sharing", self._match_texts(("METADATA", "DATA", "NONE")) and self._prev.text
                 )
             else:
                 opt = self._parse_var_from_options(self.CREATE_SEQUENCE)
                 if opt:
                     options.append(opt)
+                else:
+                    break
 
         seq.set("options", options if options else None)
         return seq
```

Comment:
This should return `None` if the sequence option is empty, otherwise you create an incorrect AST node (regardless of the generator handling the empty `exp.SequenceProperties` case).